src/relay/pass/type_infer.cc

```diff
@@ -333,6 +333,10 @@ class TypeInferencer : private ExprFunctor<Type(const Expr&)>,
     return op->op_type;
   }
 
+  Type VisitExpr_(const FatalNode* op) final {
+    return IncompleteTypeNode::make(Kind::kType);
```

Comment:
Is the idea that we want it to be possible to put Fatal inside any subexpression? I was imagining it as a `() -> ()` "statement"

Reply:
(I don't object to the idea of being able to put Fatal anywhere, but we should be clear on the semantics.)

Reply:
it is like haskell's error : a